### DISCUSSION TITLE
Code splitting with NextJS - Page sizes have increased after migrating from contexts to stores

### DISCUSSION THREAD

**Discussion by avaneeshtripathi** [pmndrs/zustand#1265] [Q&A] [answered]
I have been using React's Context API for sharing the states and actions throughout the app.

With Zustand, Since it is framework agnostic, my assumption was the code written inside a store is separate from the React's component tree.

But when i migrated all my contexts to the respective stores and replaced the usage from stores, the page sizes of my NextJS app build have increased (mostly) or remained same.

Why does this happen?
Does the page bundle still includes all the code written inside zustand stores?
And is there a way to use some code splitting which can build the Store's code as a separate chunk?

| With Context | With Zustand |
| ---- | ---- |
| <img width="511" alt="Screen Shot 2022-09-06 at 5 54 58 PM" src="https://user-images.githubusercontent.com/20407975/188653963-7bf6600e-36fa-44bd-8078-2742e26d2962.png"> | <img width="506" alt="Screen Shot 2022-09-06 at 5 55 11 PM" src="https://user-images.githubusercontent.com/20407975/188653981-55a6be47-e6c5-458e-b6d8-cdc684637dfd.png"> |

**Comment by dai-shi** [comment, marked as answer]
To my knowledge, single store like zustand is bad at code splitting. Store can't be splitted.
(and, it's one of the reasons I developed [jotai](https://github.com/pmndrs/jotai).)

You can separate actions to have better code splitting to some extent: https://github.com/pmndrs/zustand/blob/main/docs/guides/practice-with-no-store-actions.md

**Reply by avaneeshtripathi** [reply]
Thanks for clarifying. @dai-shi That helps.
